crates/ruff_cli/src/cache.rs

```diff
@@ -48,6 +44,9 @@ fn cache_key<P: AsRef<Path>>(
         .as_ref()
         .map(|path| path.as_ref().absolutize().unwrap())
         .hash(&mut hasher);
+    FileTime::from_last_modification_time(metadata).hash(&mut hasher);
```

Comment:
We now include the timestamp in the cache key. This should be more performant, since we don't have to deserialize from the cache to know that it's a miss.